tests/integration/test_solve.py

```diff
@@ -80,6 +80,109 @@ def test_solve_corner_cases():
     assert allclose(b, num.matmul(a, out))
 
 
+def test_solve_b_is_empty():
+    a = num.random.rand(1, 1)
+    b = num.atleast_2d([])
+
+    out = num.linalg.solve(a, b)
+    assert np.array_equal(b, out)
+
+
+def test_solve_dtype_int():
+    n = 8
+    a = np.random.randint(0, 10, size=(n, n), dtype="i4")
+    b = np.random.randint(0, 10, size=(n,), dtype="i4")
+
+    out = num.linalg.solve(a, b)
+
+    rtol = RTOL[out.dtype]
+    atol = ATOL[out.dtype]
+    assert allclose(
+        b, num.matmul(a, out), rtol=rtol, atol=atol, check_dtype=False
+    )
+
+
+def test_solve_with_output():
+    n = 8
+    a = np.random.rand(n, n).astype("f")
+    b = np.random.rand(n).astype("f")
+    output = np.zeros((n,)).astype("f")
+
+    out = num.linalg.solve(a, b, out=output)
+
+    rtol = RTOL[out.dtype]
+    atol = ATOL[out.dtype]
+    assert allclose(
+        b, num.matmul(a, out), rtol=rtol, atol=atol, check_dtype=False
+    )
+    assert allclose(
+        b, num.matmul(a, output), rtol=rtol, atol=atol, check_dtype=False
+    )
+
+
+class TestSolveErrors:
+    def setup_method(self):
+        self.n = 3
+        self.a = num.random.rand(self.n, self.n).astype("d")
+        self.b = num.random.rand(self.n).astype("d")
+
+    def test_a_bad_dim(self):
+        a = num.random.rand(self.n).astype("d")
```

Comment:
why is this random test value needed?

Reply:
I think random here is a way of generating a one-dimension array with shape (self.n,). It works.